src/test/java/hec/paireddata/PairedDataTest.java

```diff
@@ -0,0 +1,34 @@
+package hec.paireddata;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import java.io.File;
+import java.util.ArrayList;
+
+import hec.*;
+
+public class PairedDataTest{
+
+    @Test
+    public void A_i1d1_is_written_to_the_database() throws Exception {
+        
+        File file = new File("pairedtest.db");
+        file.delete();
+        TimeSeriesDatabase db = new JdbcTimeSeriesDatabase("pairedtest.db", true);
+        
+        PairedData table = new PairedData("test|stage/flow");
```

Comment:
This looks very simple, which is great. If we wanted to add multiple columns of data for a give row, how would we abstract that?

Reply:
Unfortunately my brain refused to work that out so it's just two columns at the moment. I did setup the addRow function to take an array list of independent input and an array list of dependent outputs that should represent that dimensions of the data set. But as I started to work out the internals I kept running into a brick wall; I'm probably just overthinking it. So currently the implementation of paired data is 1 independent and 1 dependent but the design concepts for more  are there.